pyiron_base/database/filetable.py

```diff
@@ -21,9 +22,8 @@ def __init__(self, project):
                          'username']
         self.force_reset()
 
-    @property
-    def viewer_mode(self):
-        return None
+    def _get_view_mode(self):
```

Comment:
Getters with no arguments can easily just be properties. Is there a good reason to include this method and not just make `view_mode` `@property @absractmethod`? (There might well be! I just don't see it atm.)

Reply:
Imo properties are nice for UI.  Since `IsDatabase` has a property defined this is taken care.  It's possible to make it `@property @abstractmethod ...` and then `@property ...` in the implementation, but it seems to me just one level of indirection and more typing.  It's cosmetic only, so I'm easily overruled.

Reply:
Hmm, I think we've talked past each other although I don't know who's misunderstanding.

I'm trying to propose one fewer level of indirection by setting `IsDatabase.view_mode` as an `@property@abstractmethod` and removing `@abstractmethod def _get_view_mode` entirely. In this case I guess it's true that devs need to remember `@property` when overriding, is that your point? I guess then it's not a cosmetic difference so much as a cost/benefit of an extra line in the overrides vs one less method (and one less layer of indirection). I lean towards more typing and less layers of indirection, but only weakly.

Reply:
No, I think I got some things mixed up, as you said the only consequence should be more typing, which is ok with me.  I'll change accordingly.

Reply:
Looking at the code again, one advantage of having a proper property and abstract getter/setters is that you only need to document the property once.

Reply:
> Looking at the code again, one advantage of having a proper property and abstract getter/setters is that you only need to document the property once.

I'm afraid it's unclear to me whether this is advocating for or against removing `def _get_view_mode` in favour of `@property@abstractmethod def view_mode`. IMO the documentation doesn't come into play as the attribute should be documented at the class level. I don't specifically object to property-level docstrings, but since non-property attributes need to get documented at the class level anyhow, I just figure it's unnecessary work.

Honestly though, I'm still torn between whether it's better to demand the extra typing or to introduce an extra layer of abstraction...